skglm/tests/test_group.py

```diff
@@ -160,6 +161,66 @@ def test_intercept_grouplasso():
     np.testing.assert_allclose(model.intercept_, w[-1], atol=1e-5)
 
 
+@pytest.mark.parametrize("rho", [1e-1, 1e-2])
+def test_equivalence_logreg(rho):
+    n_samples, n_features = 30, 50
+    rnd = np.random.RandomState(1123)
+    X, y, _ = make_correlated_data(n_samples, n_features, random_state=rnd)
+    y = np.sign(y)
+
+    grp_indices, grp_ptr = grp_converter(1, n_features)
+    weights = np.ones(n_features)
+    alpha_max = norm(X.T @ y, ord=np.inf) / (2 * n_samples)
+    alpha = rho * alpha_max / 10.
+
+    log_group = LogisticGroup(grp_ptr=grp_ptr, grp_indices=grp_indices)
+    group_penalty = WeightedGroupL2(
+        alpha=alpha, grp_ptr=grp_ptr,
+        grp_indices=grp_indices, weights=weights)
+
+    log_group = compiled_clone(log_group, to_float32=X.dtype == np.float32)
+    group_penalty = compiled_clone(group_penalty)
+    w = GroupBCD(tol=1e-12).solve(X, y, log_group, group_penalty)[0]
+
+    sk_logreg = LogisticRegression(penalty='l1', C=1/(n_samples * alpha),
+                                   fit_intercept=False, tol=1e-12, solver='liblinear')
+    sk_logreg.fit(X, y)
+
+    np.testing.assert_allclose(sk_logreg.coef_.flatten(), w, atol=1e-6, rtol=1e-5)
+
+
+@pytest.mark.parametrize("n_groups, rho", [[15, 1e-1], [25, 1e-2]])
+def test_group_logreg(n_groups, rho):
+    n_samples, n_features, shuffle = 30, 60, True
+    random_state = 123
+
+    X, y, _ = make_correlated_data(n_samples, n_features, random_state=random_state)
+    y = np.sign(y)
+
+    np.random.seed(random_state)
```

Comment:
usually this is frowned upo (it resets the seed globally, that's a heavy side effect) 

use rng = sklearn.utils.check_random_state(random_state), then rng.randn